metaflow/plugins/kubernetes/kubernetes_decorator.py

```diff
@@ -175,13 +177,14 @@ def __init__(self, attributes=None, statically_defined=False):
             except (NameError, ImportError):
                 pass
 
-        # parse the CPU, memory, disk, values from the KUBERNETES_RESOURCES environment variable (you would need to export the METAFLOW_KUBERNETES_RESOURCES environment variable with the desired values before running the flow)
-        # find the values in the format "cpu=1,memory=4096,disk=10240", if not set, the default values described above are used
-        if KUBERNETES_RESOURCES:
-            for resource in KUBERNETES_RESOURCES.split(","):
-                key, value = resource.split("=")
-                if key in self.attributes:
-                    self.attributes[key] = value
+        # parse the CPU, memory, disk, values from the KUBERNETES_ environment variable (you would need to export the METAFLOW_KUBERNETES_CPU, METAFLOW_KUBERNETES_MEMORY and/or METAFLOW_KUBERNTES_DISK environment variable with the desired values before running the flow)
+        # find the values from the environment variables, if not set, the default values as defined above are used
+        if KUBERNETES_CPU:
+            self.attributes["cpu"] = KUBERNETES_CPU
+        if KUBERNETES_MEMORY:
+            self.attributes["memory"] = KUBERNETES_MEMORY
+        if KUBERNETES_DISK:
+            self.attributes["disk"] = KUBERNETES_DISK
```

Comment:
One more change I would suggest is having a similar check to existing env var setters, where the default does not override a value that the user has supplied.
as an example:

```
METAFLOW_KUBERNETES_CPU=3

@kubernetes(cpu=2)
@step
```
expectation would be that cpu gets set to 2, not 3.

note though that the check can not be identical to the existing ones, as we have proper default values for some of the attributes, so f.ex. `self.attributes["cpu"]` will always be set.

Reply:
found the way to validate that, thanks